jwst/datamodels/library.py

```diff
@@ -246,28 +246,51 @@ def get_crds_parameters(self):
                 )
                 idx = 0
 
-            # find model in _loaded_models, temp_filenames, or asn_dir
-            if self._on_disk:
-                if idx in self._temp_filenames:
-                    # if model has been modified, find its temp filename
-                    filename = self._temp_filenames[idx]
-                else:
-                    # otherwise, find the filename in the asn_dir
-                    member = self._members[idx]
-                    filename = Path(self._asn_dir) / member["expname"]
+            return self.read_metadata(idx)
+
+    def read_metadata(self, idx):
+        """
+        Read metadata for a model at the given index.
+
+        Parameters
+        ----------
+        idx : int
+            Index of the model in the library.
+
+        Returns
+        -------
+        dict
+            The metadata dictionary for the model.
+        """
+        if not self._open:
```

Comment:
Can we lift the restriction here allowing this to be called:
- on closed libraries
- more than once per member

Those restrictions are in place to avoid complications from model modifications. Are those supported by this method?

Reply:
sure, that works, I just tried to make this look a lot like "borrow".  But no, model modification is not supported so this should not be necessary.